catalog/core/catalog-core-standardframework/src/test/java/org/codice/ddf/catalog/resource/download/impl/ResourceDownloadTest.java

```diff
@@ -0,0 +1,261 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.resource.download.impl;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+
+import javax.management.MBeanException;
+import javax.management.MBeanRegistrationException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.StandardMBean;
+
+import org.codice.ddf.catalog.resource.cache.ResourceCacheServiceMBean;
+import org.codice.ddf.catalog.resource.download.ResourceDownloadMBean;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.resource.ResourceNotFoundException;
+import ddf.catalog.resource.ResourceNotSupportedException;
+import ddf.catalog.resource.download.ReliableResourceDownloadManager;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ResourceDownloadTest {
+    private static final String METACARD_ID = "57a4b894e13a455b8cccb87cec778b58";
+
+    private static final String SOURCE_ID = "ddf.distribution";
+
+    private static final String SUCCESS_MESSAGE_TEMPLATE = "Successfully started download of resource associated with metacard id %s from source %s";
+
+    @Mock
+    private CatalogFramework mockCatalogFramework;
+
+    @Mock
+    private ReliableResourceDownloadManager mockDownloadManager;
+
+    @Mock
+    private ResourceCacheServiceMBean mockResourceCacheServiceMBean;
+
+    @Mock
+    private ResourceResponse mockResourceResponse;
+
+    @Mock
+    private MBeanServer mockMBeanServer;
+
+    private ObjectName resourceDownloadObjectName;
+
+    @Before
+    public void before() throws Exception {
+        resourceDownloadObjectName = new ObjectName(ResourceDownloadMBean.OBJECT_NAME);
+    }
+
+    @Test(expected = MBeanException.class)
+    public void downloadToCacheWhenNullResourceResponse() throws Exception {
+        // Setup
+        setupMockResourceCacheServiceMBean(true);
+        setupMockDownloadManager(true);
+        setupMockCatalogFramework(null, null);
+        ResourceDownloadMBean resourceDownloadMBean = createResourceDownloadMBean();
+
+        // Perform Test
+        resourceDownloadMBean.downloadToCache(SOURCE_ID, METACARD_ID);
```

Comment:
Feels like we should have a few `verify()` calls on the mock objects in those tests.